lazy-pages/src/sys.rs

```diff
@@ -105,24 +105,35 @@ pub unsafe fn user_signal_handler(info: ExceptionInfo) -> Result<(), Error> {
 
     region::protect(unprot_addr as *mut (), unprot_size, Protection::READ_WRITE)?;
 
-    LAZY_PAGES_INFO.with(|lazy_pages_info| {
-        let mut pages_info = lazy_pages_info.borrow_mut();
+    LAZY_PAGES_CONTEXT.with(|ctx| {
+        let mut ctx = ctx.borrow_mut();
         for idx in 0..gear_pages_num as u32 {
-            let page = LazyPage::from(gear_page) + idx;
+            let page = gear_page + idx.into();
+
+            if !ctx.lazy_pages.remove(&page) {
+                return Err(Error::LazyPageNotExistForSignalAddr(mem, page));
+            }
 
-            let hash_key_in_storage = pages_info.remove(&page).ok_or(LazyPageError::UnknownInfo(page))?;
             let ptr = (unprot_addr as *mut u8).add(idx as usize * gear_ps);
             let buffer_as_slice = std::slice::from_raw_parts_mut(ptr, gear_ps);
```

Comment:
do all safety prerequisites correct here? should we extra check it?

Reply:
It's not rust safe) but it is used inside signal handler (which is already very unsafe), which must works with raw memory)

Reply:
Nvm, I meant requirements to avoid ub from [here](https://doc.rust-lang.org/std/slice/fn.from_raw_parts_mut.html)